addons/hr/models/mail_activity_plan.py

```diff
@@ -36,3 +38,26 @@ def _check_compatibility_with_model(self):
     def _compute_department_id(self):
         for plan in self.filtered(lambda plan: plan.res_model != 'hr.employee'):
             plan.department_id = False
+
+    @api.depends('template_ids.summary')
+    def _compute_assignation_summary(self):
+        if 'sort_by_responsible' in self.env.context and self.env.context.get('sort_by_responsible'):
+            self.assignation_summary = ''
+            for plan in self.filtered('template_ids'):
+                summaries = defaultdict()
+                for template in plan.template_ids:
+                    summary = template.activity_type_id.name + (f": {template.summary}" if template.summary else '')
+                    if template.responsible_type in summaries:
+                        summaries[template.responsible_type].append(summary)
+                    else:
+                        summaries[template.responsible_type] = [summary]
+                if summaries:
+                    for key, template_type in summaries.items():
+                        plan.assignation_summary = plan.assignation_summary + Markup('<ul>%(key)s<ul>%(template)s</ul></ul>') % {
+                            'key': key,
+                            'template': Markup().join(Markup('<li>%s</li>') % summary for summary in template_type)
+                        }
+                else:
+                    plan.assignation_summary = ''
+        else:
+            return super()._compute_assignation_summary()
```

Comment:
? 
```suggestion
    @api.depends('template_ids.summary')
    def _compute_assignation_summary(self):
        if not self.env.context.get('sort_by_responsible'):
            return super()._compute_assignation_summary()

        self.assignation_summary = ''
        for plan in self:
            if not plan.template_ids:
                continue
            summaries = defaultdict(list)
            for template in plan.template_ids:
                summary = template.activity_type_id.name + (f": {template.summary}" if template.summary else '')
                summaries[template.responsible_type].append(summary)
            plan.assignation_summary = ''.join(Markup('<ul>%(key)s<ul>%(template)s</ul></ul>') % {
                'key': key,
                'template': Markup().join(Markup('<li>%s</li>') % summary for summary in template_type)
            } for key, template_type in summaries.items()
```

Reply:
I tried to understand my own code and you're right it's so strange :joy: